includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -641,6 +645,8 @@ public function prepare_source( $user_id, $force_save_source = false ) {
 		if ( ! $customer_id ) {
 			$customer->set_id( $customer->create_customer() );
 			$customer_id = $customer->get_id();
+		} else {
+			$customer->update_customer();
```

Comment:
Why do we need to call `update_customer()` here? From what I can see, we aren't sending the new `source` to stripe, and what we're getting back from the API call in `update_customer` is a customer object (https://stripe.com/docs/api/customers/update) with what looks like no differences to what we had before. The customer ID is already going to be correct since we set it earlier using `$customer->set_id( $existing_customer_id );`.

Reply:
> Why do we need to call update_customer() here? From what I can see, we aren't sending the new source to stripe, and what we're getting back from the API call in update_customer is a customer object (https://stripe.com/docs/api/customers/update) with what looks like no differences to what we had before.

Most of the time this call is redundant, yes. But customers might also fix/change their names, as well as their email address while changing the credit card they entered. In this case, the customer object needs to be updated.

Now, at the time `update_customer` and the `POST` request are executed, we do not have the data about the customer. In order to avoid this call, we'd first need another `GET` and a comparison of the data in Stripe and the one we have. This means that if the details _did_ change, we'd have two API calls (get and update), instead of just having this one call (update) anyway.

Reply:
Ok, I see  👍 Nice.